rest/deployments.go

```diff
@@ -366,35 +364,26 @@ func (s *Server) getDeploymentHandler(w http.ResponseWriter, r *http.Request) {
 }
 
 func (s *Server) listDeploymentsHandler(w http.ResponseWriter, r *http.Request) {
-	kv := s.consulClient.KV()
 	ctx := r.Context()
-	depPaths, _, err := kv.Keys(consulutil.DeploymentKVPrefix+"/", "/", nil)
+	deploymentsIDs, err := deployments.GetDeploymentsIDs(ctx)
 	if err != nil {
 		log.Panic(err)
 	}
-	if len(depPaths) == 0 {
+	if len(deploymentsIDs) == 0 {
 		w.WriteHeader(http.StatusNoContent)
 		return
 	}
 
 	deps := make([]Deployment, 0)
-	depPrefix := consulutil.DeploymentKVPrefix + "/"
-	for _, depPath := range depPaths {
-		deploymentID := strings.TrimRight(strings.TrimPrefix(depPath, depPrefix), "/ ")
+	for _, deploymentID := range deploymentsIDs {
 		status, err := deployments.GetDeploymentStatus(ctx, deploymentID)
 		if err != nil {
 			if deployments.IsDeploymentNotFoundError(err) {
-				// Deployment is not found : we force deletion and ignore it
-				go func() {
-					log.Debugf("Force purge inconsistent deployment with ID:%q", deploymentID)
-					if _, err := s.tasksCollector.RegisterTask(deploymentID, tasks.TaskTypeForcePurge); err != nil {
-						log.Printf("Failed to force purge deployment with ID:%q due to error:%+v", deploymentID, err)
-					}
-				}()
+				// Deployment is not found : we force rapport error and ignore it
+				log.Printf("[WARNING] deployment %q is inconsistent, igoring it from deployments list. You please investigate and retport this issue.")
```

Comment:
Missing deployement ID in argument of Printf

Reply:
good catch